internal/auth/providers/amazon_cognito_admin.go

```diff
@@ -0,0 +1,270 @@
+package providers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/buzzfeed/sso/internal/auth/circuit"
+	log "github.com/buzzfeed/sso/internal/pkg/logging"
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+	"github.com/datadog/datadog-go/statsd"
+)
+
+type CognitoAdminProvider interface {
+	ListMemberships(groupName string) (members []string, err error)
+	CheckMemberships(userName string) (inGroups []string, err error)
+	GlobalSignOut(session *sessions.SessionState) (err error)
+	GetUserInfo(accessToken *string) (userInfo *cognitoidentityprovider.GetUserOutput, err error)
+}
+
+type CognitoAdminService struct {
+	adminService *cognitoidentityprovider.CognitoIdentityProvider
+	StatsdClient *statsd.Client
+	cb           *circuit.Breaker
+	userPoolID   *string
+}
+
+func getCognitoIdentityProvider(id, secret, region string) (*cognitoidentityprovider.CognitoIdentityProvider, error) {
+	logger := log.NewLogEntry()
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region),
+		Credentials: credentials.NewStaticCredentials(
+			id,
+			secret,
+			""),
+	})
+	if err != nil {
+		logger.WithError(err).Fatal("error creating aws session")
+		return nil, err
+	}
+	idp := cognitoidentityprovider.New(sess)
+
+	return idp, nil
+}
+
+//TODO: we could potentially remove this in favour of the GetUserInfo cognito provider method
```

Comment:
Just to highlight this `TODO`. We could remove it, though it's also quite nice having the admin module contain everything needed?

Reply:
Looks like we implemented this by calling the admin service anyway -- so it seems reasonable to just use that and we can delete this func.